client/network/src/discovery.rs

```diff
@@ -100,7 +101,7 @@ pub struct DiscoveryConfig {
 	discovery_only_if_under_num: u64,
 	enable_mdns: bool,
 	kademlia_disjoint_query_paths: bool,
-	protocol_ids: HashSet<ProtocolId>,
+	kademlia_protocol_id: Option<ProtocolId>,
```

Comment:
IMO this should not be made optional. We should take this as input to `new`.

The comment above the struct is also stale.

Reply:
`Discovery` is only about Kademlia or not? At least by looking at the code, I don't see that much different stuff going on. If you don't want any discovery at all, you could probably just turn it off completely?

Reply:
`Discovery` is also about mDNS and bootnodes, so it's in theory possible to use it without Kademlia. What I don't know is whether it makes sense doing this (some LAN-only blockchains with mDNS discovery?)

Reply:
Hmm yeah okay, I had overseen `mdns`. Nevertheless, we should update the comment above the struct.

Reply:
> Nevertheless, we should update the comment above the struct.

Done.